### PR TITLE
Adding c# examples for call, dequeue, and redirect

### PR DESCRIPTION
**Pull request by wanjunsli**
C# code snippets are currently missing on TaskRouter Update a Reservation REST docs for Call, Dequeue, Redirect instructions.

### CODE REVIEW

**Comment by mcelicalderon** [comment]
Build is failing. This probably means that I deleted those snippets due to a bug in the c# library. I'll take a look.

**Comment by wanjunsli** [comment]
Sounds good! Let me know if there is something you need me to do to get the build to pass.
I tested these snippets too, so they should be good to go.

File: rest/taskrouter/reservations/dequeue/example-1.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    var client = new TaskRouterClient(AccountSid, AuthToken);
+
+    // Update a Reservation with a Call instruction
+    Reservation reservation = client.GetReservation(WorkspaceSid, TaskSid, ReservationSid);
+    Console.WriteLine(reservation.ReservationStatus);
+    Console.WriteLine(reservation.WorkerName);
+
+    client.UpdateReservation(WorkspaceSid, "Tasks", TaskSid, ReservationSid,
+        instruction: "dequeue",
+        dequeueTo: "+12345678901",
```
Comment (mcelicalderon) [inline rest/taskrouter/reservations/dequeue/example-1.cs]: The other languages don't update this parameter and it is optional. Please remove it to maintain consistency across languages.

File: rest/taskrouter/reservations/call/example-1.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    var client = new TaskRouterClient(AccountSid, AuthToken);
+
+    // Update a Reservation with a Call instruction
+    Reservation reservation = client.GetReservation(WorkspaceSid, TaskSid, ReservationSid);
+    Console.WriteLine(reservation.ReservationStatus);
+    Console.WriteLine(reservation.WorkerName);
+
+    client.UpdateReservation(WorkspaceSid, "Tasks", TaskSid, ReservationSid,
+        instruction: "call",
+        callTo: "+12345678901",
```
Comment (mcelicalderon) [inline rest/taskrouter/reservations/call/example-1.cs]: Same here. Other languages don't update this. And please add `callAccept: "true"`

File: rest/taskrouter/reservations/redirect/example-1.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    string ReservationSid = "WRXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+    var client = new TaskRouterClient(AccountSid, AuthToken);
+
+    // Update a Reservation with a Call instruction
+    Reservation reservation = client.GetReservation(WorkspaceSid, TaskSid, ReservationSid);
+    Console.WriteLine(reservation.ReservationStatus);
+    Console.WriteLine(reservation.WorkerName);
+
+    client.UpdateReservation(WorkspaceSid, "Tasks", TaskSid, ReservationSid,
+        instruction: "redirect",
```
Comment (mcelicalderon) [inline rest/taskrouter/reservations/redirect/example-1.cs]: Please change to `Redirect`

**Comment by mcelicalderon** [comment]
@wanjunsli Other than the comments I left, this looks good! Build is already passing after your second commit. Please fix the errors and squash into one descriptive message commit. Thanks for the new snippets!

**Comment by mcelicalderon** [comment]
LGTM!

**Pull request merged**